### PR TITLE
Propagate the `cython --version` error

### LINKED ISSUE #2000: `FindCython` ignores error exit code and parses stderr for version numbers
**Describe the bug**
`FindCython.cmake` parses error messages for version numbers and then uses them as the detected version for cython. This is a problem for pyenv users who might have a shim for cython which produces an error of this format:

```
pyenv: cython: command not found

The `cython' command exists in these Python versions:
  2.7.18/envs/nrn-dev-py2
  3.8.6
  3.8.6/envs/gaperr
  3.9.0/envs/nrn-dev
  gaperr
  nrn-dev
  nrn-dev-py2

Note: See 'pyenv help global' for tips on allowing both
      python2 and python3 to be found.
```

**To Reproduce**
1. Create a script called `cython` on a location detected by CMake
2. Make it print "Super critical error, you need this-other-library 3.3.3 or higher to run Cython." to stderr and exit with an error code
3. `FindCython` detects cython version 3.3.3

**Expected behavior**
FindCython should know this is an error, and not parse it for version numbers.

### PR DESCRIPTION
**Pull request by Helveg**
Fixes #2000 by propagating the error that `cython --version` throws

### CODE REVIEW

**Review by gtrensch — APPROVED** [review]
Thanks for this contribution. It looks good to me :+1: .
Approving.

**Comment by Helveg** [comment]
Will do, it will require some debugging as this could only occur if Cython returns an error code to `cython --version` or if it prints it output to `stderr`, both seem unlikely. 😕

**Comment by clinssen** [comment]
``cython --version`` apparently prints to stderr, not stdout. D'oh!

**Comment by Helveg** [comment]
O_O ok, will fix, the exit code is 0 though I hope?

File: cmake/FindCython.cmake [hunk 1]
```diff
@@ -43,13 +43,22 @@ if ( NOT CYTHON_EXECUTABLE STREQUAL "CYTHON_EXECUTABLE-NOTFOUND" )
...
       OUTPUT_VARIABLE CYTHON_VAR_OUTPUT
-      ERROR_VARIABLE CYTHON_VAR_OUTPUT
+      ERROR_VARIABLE CYTHON_ERR_OUTPUT
       OUTPUT_STRIP_TRAILING_WHITESPACE
   )
   if ( RESULT EQUAL 0 )
+    if ( "${CYTHON_VAR_OUTPUT}" STREQUAL "" )
+      # If `cython --version` stdout is empty but the
+      # return code is 0 (no error), Cython might have
+      # printed to stderr instead.
```
Comment (gtrensch) [inline cmake/FindCython.cmake:53]:
```suggestion
      # In cython v0.29.3 the version string is written to stderr and not to stdout, as one would expect.
```

Comment (gtrensch) [reply cmake/FindCython.cmake:53]: I've changed the comment to be more precise what the issue is.

**Review by gtrensch — CHANGES_REQUESTED** [review]
Thanks for the update! 
I ran some CMake tests forcing some errors to verify the robustness of the solution. It works! Problem is, as @clinssen already mentioned, that cython writes the version string to stderr and not stdout, as one could expect. This might perhaps change in future. I've suggested to adjusted the comment to be more precise what the issue is. Please check. I would then approve this PR.

**Comment by Helveg** [comment]
Merged the suggestion

**Review by gtrensch — APPROVED** [review]
Approved!

**Review by clinssen — APPROVED** [review]

**Pull request merged**